tests/network/test_rabbitmq.py

```diff
@@ -0,0 +1,83 @@
+DEFAULT_QUEUE = config.DEFAULT_QUEUE
+
+
+def test_rmq_publish_2_messages(rabbitmq_publisher):
+    test_headers = {"Test Header": "Test Value"}
+    test_properties = BasicProperties(
+        app_id='webapi-processing-broker',
+        content_type='application/json',
+        headers=test_headers
+    )
+    rabbitmq_publisher.publish_to_queue(
+        queue_name=DEFAULT_QUEUE,
+        message="RabbitMQ test 123",
+        exchange_name=DEFAULT_EXCHANGER_NAME,
+        properties=test_properties
+    )
+    rabbitmq_publisher.publish_to_queue(
+        queue_name=DEFAULT_QUEUE,
+        message="RabbitMQ test 456",
+        exchange_name=DEFAULT_EXCHANGER_NAME,
+        properties=test_properties
+    )
+    assert rabbitmq_publisher.message_counter == 2
+
+
+def test_rmq_consume_2_messages(rabbitmq_consumer):
+    # Consume the 1st message
+    method_frame, header_frame, message = rabbitmq_consumer.get_one_message(
+        queue_name=DEFAULT_QUEUE,
+        auto_ack=True
+    )
+    assert method_frame.delivery_tag == 1  # 1st delivered message to this queue
+    assert method_frame.message_count == 1  # messages left in the queue
+    assert method_frame.redelivered is False
+    assert method_frame.exchange == DEFAULT_EXCHANGER_NAME
+    assert method_frame.routing_key == DEFAULT_QUEUE
+    # It's possible to assert header_frame the same way
+    assert message.decode() == "RabbitMQ test 123"
+
+    # Consume the 2nd message
+    method_frame, header_frame, message = rabbitmq_consumer.get_one_message(
+        queue_name=DEFAULT_QUEUE,
+        auto_ack=True
+    )
+    assert method_frame.delivery_tag == 2  # 2nd delivered message to this queue
+    assert method_frame.message_count == 0  # messages left in the queue
+    assert method_frame.redelivered is False
+    assert method_frame.exchange == DEFAULT_EXCHANGER_NAME
+    assert method_frame.routing_key == DEFAULT_QUEUE
+    # It's possible to assert header_frame the same way
+    assert message.decode() == "RabbitMQ test 456"
+
+
+def test_rmq_publish_ocrd_message(rabbitmq_publisher):
+    ocrd_processing_message = {
+        "job_id": "Test_job_id",
+        "workflow_id": "Test_workflow_id",
+        "workspace_id": "Test_workspace_id"
+    }
+    message_bytes = dumps(ocrd_processing_message)
+    rabbitmq_publisher.publish_to_queue(
+        queue_name=DEFAULT_QUEUE,
+        message=message_bytes,
+        exchange_name=DEFAULT_EXCHANGER_NAME,
+        properties=None
+    )
+
+
+def test_rmq_consume_ocrd_message(rabbitmq_consumer):
+    method_frame, header_frame, message = rabbitmq_consumer.get_one_message(
+        queue_name=DEFAULT_QUEUE,
+        auto_ack=True
+    )
+    assert method_frame.message_count == 0  # messages left in the queue
+    decoded_message = loads(message)
+    assert decoded_message["job_id"] == "Test_job_id"
+    assert decoded_message["workflow_id"] == "Test_workflow_id"
+    assert decoded_message["workspace_id"] == "Test_workspace_id"
```

Comment:
It seems to me some of the tests depend on each other or rather that they must be executed in order. It might be better to combine them into one test because I think the order of the test execution might not be from top to bottom always

Reply:
You are right. I will combine these.